common/logisticspipes/gui/guidebook/GuiGuideBook.java

```diff
@@ -0,0 +1,779 @@
+package logisticspipes.gui.guidebook;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.client.renderer.BufferBuilder;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.MathHelper;
+
+import logisticspipes.items.ItemGuideBook;
+import logisticspipes.utils.string.StringUtils;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.lwjgl.opengl.GL11;
+
+import logisticspipes.LPConstants;
+import logisticspipes.gui.guidebook.book.MenuItem;
+import logisticspipes.utils.GuideBookContents;
+
+public class GuiGuideBook extends GuiScreen {
+
+	/*
+	 * TODO - Back button or bookmarks, whichever is more wanted
+	 *		- Image support
+	 *		- Text personalization support (bold, italics, underline)
+	 * 			(I'd suggest a subset of markdown format for that, which has just the italic/bold/underline syntax, so that it can be expanded later if needed- @farfetched)
+	 * 		- Link?
+	 * 		- Search bar?
+	 * 		- NEI/JEI integration - Information page.
+	 */
+
+	/*
+	 * Z Levels:
+	 * Tooltip          : 20
+	 * Title and Buttons: 15
+	 * Background frame : 10
+	 * Text             : 5
+	 * Background       : 0
+	 * Background tint  : -5
+	 */
+
+	private boolean loadedNBT = false;
+	private final int z$tooltip = 20;      // Tooltip z
+	private final int z$titleButtons = 15; // Title and Buttons Z
+	private final int z$frame = 10;        // Frame Z
+	private final int z$text = 5;          // Text/Information Z
+	private final int z$background = 0;    // Background Z
+
+	private static final ResourceLocation GUI_BOOK_TEXTURE = new ResourceLocation(LPConstants.LP_MOD_ID, "textures/gui/guide_book.png");
+
+	private GuideBookContents gbc;
+	//////// Buttons
```

Comment:
Please don't fancy style your comments, but keep the usual coding style

Reply:
Sure